src/basic_recipes/pie.jl

```diff
@@ -17,14 +19,25 @@ Creates a pie chart from the given `values`.
     inner_radius = 0
     "The angular offset of the first pie segment from the (1, 0) vector in radians."
     offset = 0
+    "The offset of each pie segment from the center along the radius"
+    offset_radius = 0
     MakieCore.mixin_generic_plot_attributes()...
 end
 
-function plot!(plot::Pie)
+convert_arguments(::Type{<:Pie}, values::Vector) = (0, 0, values)
+convert_arguments(::Type{<:Pie}, point::Point, values::Vector) = (point[1], point[2], values)
+convert_arguments(::Type{<:Pie}, point, values::Vector) = (getindex.(point, 1), getindex.(point, 2), values)
+convert_arguments(::Type{<:Pie}, xs, ys, values::Vector) = (xs, ys, values)
```

Comment:
Some of these types need to be more general to work with e.g. ranges and Vec, and some should be more restricted so conversions fail on bad inputs. I think for compile times it's also good to normalize the output type of convert_arguments more.

```julia
convert_arguments(PT::Type{<:Pie}, values::RealVector) = convert_arguments(PT, [0.0], [0.0], values)
convert_arguments(PT::Type{<:Pie}, point::VecTypes{2}, values::RealVector) = convert_arguments(PT, point[1], point[2], values)
convert_arguments(PT::Type{<:Pie}, ps::AbstractVector{<: VecTypes{2}}, values::RealVector) = convert_arguments(PT, getindex.(point, 1), getindex.(point, 2), values)
convert_arguments(PT::Type{<:Pie}, x::Real, y::Real, values::RealVector) = convert_arguments(PT, [x], [y], values)
convert_arguments(::Type{<:Pie}, xs::RealVector, ys::RealVector, values::RealVector) = (floatconvert(xs), floatconvert(ys), floatconvert(values))
```

For reference, `RealVector` is just `AbstractVector{<: Real}` which includes ranges, `VecTypes` include Vec and Tuples and `floatconvert` will convert inputs to either Float32 or Float64 vectors.

With this xs, ys, and values should always end up as Vectors. If it's easier in the recipe you can also also swap out `[x]` for something like `[x for _ in values]`.